src/screens/splash.rs

```diff
@@ -135,8 +173,34 @@ fn tick_splash_timer(time: Res<Time>, mut timer: ResMut<SplashTimer>) {
     timer.0.tick(time.delta());
 }
 
-fn check_splash_timer(timer: ResMut<SplashTimer>, mut next_screen: ResMut<NextState<Screen>>) {
-    if timer.0.just_finished() {
-        next_screen.set(Screen::Loading);
+fn check_splash_timer(
+    mut commands: Commands,
+    mut timer: ResMut<SplashTimer>,
+    mut next_screen: ResMut<NextState<Screen>>,
+    mut splash_images: ResMut<SplashScreenImageList>,
+    containers: Query<Entity, With<SplashScreenContainer>>,
+) {
+    if !timer.0.just_finished() {
+        return;
     }
+
+    // Try to get the next splash image. If there isn't one, we move on to the next screen.
+    let Some(next_splash_image) = splash_images.0.pop_front() else {
+        // There are no more splash screens, exit to the loading screen.
+        next_screen.set(Screen::Loading);
+        return;
+    };
+
+    let Ok(container) = containers.get_single() else {
+        return;
+    };
+
+    commands
+        .entity(container)
+        .despawn_descendants()
+        .with_children(|parent| {
+            parent.spawn(splash_image_bundle(next_splash_image));
+        });
+
+    timer.0.reset();
```

Comment:
Change `TimerMode::Once` -> `TimerMode::Repeating`, and then this line isn't needed.